types/types.go

```diff
@@ -23,6 +24,20 @@ type Repository interface {
 	Image(ref string) Image // ref == image name w/o registry part
 }
 
+// ImageSource is a service, possibly remote (= slow), to download components of a single image.
+type ImageSource interface {
+	GetManifest() (manifest []byte, unverifiedCanonicalDigest string, err error)
+	GetLayer(digest string) (io.ReadCloser, error)
+	GetSignatures() ([][]byte, error)
+}
+
+// ImageDestination is a service, possibly remote (= slow), to store components of a single image.
+type ImageDestination interface {
+	PutManifest([]byte) error
+	PutLayer(digest string, stream io.ReadCloser) error
+	PutSignatures(signatures [][]byte) error
+}
```

Comment:
I'm not really fond to those interfaces (and their naming). Why a single interface/object can't do both (get|put)?